src/fides/api/ops/models/privacy_experience.py

```diff
@@ -165,26 +297,302 @@ def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyExperience:
+
+    Our current primary check is that opt-in and notice only notices have to be displayed in
+    an overlay delivered by a banner.
+    """
+    return (
+        component_type == ComponentType.overlay
+        and delivery_mechanism == DeliveryMechanism.link
+        and bool(
+            get_privacy_notices_by_region_and_component(
+                db, region, ComponentType.overlay  # type: ignore[arg-type]
+            )
+            .filter(
+                PrivacyNotice.consent_mechanism.in_(
+                    [ConsentMechanism.opt_in, ConsentMechanism.notice_only]
+                )
+            )
+            .first()
+        )
+    )
+
+
+def upsert_privacy_experiences_after_config_update(
+    db: Session,
+    experience_config: PrivacyExperienceConfig,
+    regions: List[PrivacyNoticeRegion],
+) -> Tuple[
+    List[PrivacyNoticeRegion], List[PrivacyNoticeRegion], List[PrivacyNoticeRegion]
+]:
+    """
+    Keeps Privacy Experiences in sync with ExperienceConfig changes.
+    Create or update PrivacyExperiences for the regions we're attempting to link to the
+    ExperienceConfig.  If the region cannot be linked, or if it is currently attached
+    and shouldn't be, skip or remove that region.
+    """
+    linked_regions: List[PrivacyNoticeRegion] = []  # Regions that were linked
+    unlinked_regions: List[PrivacyNoticeRegion] = []  # Regions that were unlinked
+    skipped_regions: List[PrivacyNoticeRegion] = []  # Regions that were skipped
+
+    for region in regions:
+        (
+            overlay_experience,
+            privacy_center_experience,
+        ) = PrivacyExperience.get_experiences_by_region(db, region)
+
+        existing_experience: Optional[PrivacyExperience] = (
+            overlay_experience
+            if experience_config.component == ComponentType.overlay
+            else privacy_center_experience
+        )
+
+        # This should be caught ahead of time validation at the API level, but if this is called directly,
+        # prioritize keeping the Experiences in a valid state at all times.
+        if config_incompatible_with_region(
+            db,
+            experience_config.component,  # type: ignore[arg-type]
+            experience_config.delivery_mechanism,  # type: ignore[arg-type]
+            region,
+        ):
+            if (
+                existing_experience
+                and existing_experience.experience_config_id
+                and existing_experience.experience_config_id == experience_config.id
+            ):
+                # If existing experience is already linked to a config that is now invalid, unlink it.
+                existing_experience.unlink_privacy_experience_config(db)
+                unlinked_regions.append(region)
+            else:
+                # Skip linking experience altogether to config for which it would be incompatible
+                skipped_regions.append(region)
+            continue
```

Comment:
In practice, I added validation to the endpoints that create and update an experience config to fail early if we're attempting to make a change that would put Experiences in a bad state, so I wouldn't expect this to get hit.

But I think it's important to prioritize that Experiences always match the Notices, so I'm leaving this second level here as a fallback.